ogr/services/gitlab/project.py

```diff
@@ -425,9 +425,27 @@ def create_issue(
         body: str,
         private: Optional[bool] = None,
         labels: Optional[List[str]] = None,
+        assignees: Optional[List[str]] = None,
     ) -> Issue:
+
+        ids = []
+        if assignees:
+            for user in assignees:
+                try:
+                    ids.append(
+                        str(
+                            self.service.gitlab_instance.users.list(username=user)[0].id
+                        )
+                    )
+                except gitlab.GitlabAPIException:
+                    raise GitlabAPIException(f"Unable to find '{user}' username")
```

Comment:
Looks weird but I can't think of a better code for this..;)

Maybe this?

```suggestion
        ids = []
        for user in assignees or []:
            try:
                ids.append(
                    str(
                        self.service.gitlab_instance.users.list(username=user)[0].id
                    )
                )
            except gitlab.GitlabAPIException:
                raise GitlabAPIException(f"Unable to find '{user}' username")
```

Reply:
Or what about this (looks like a really interesting problem..;):

```suggestion
        try:
            ids = [
                str(self.service.gitlab_instance.users.list(username=user)[0].id)
                for user in assignees or []
            ]
        except gitlab.GitlabAPIException:
            # ok, we should not use `user` here
            raise GitlabAPIException(f"Unable to find '{user}' username")
```

Reply:
Last note..;)

Ca we check that the exception really means the not found user? (More specific type and/or content of the exception?)

Reply:
The API does not throw any error if the user is not found.

Reply:
But will throw out of index - `str(self.service.gitlab_instance.users.list(username=user)[0]`

Reply:
How about this -

```
        ids = []
        for user in assignees or []:
            users_list = self.service.gitlab_instance.users.list(username=user)

            if users_list == []:
                raise GitlabAPIException(f"Unable to find '{user}' username")

            ids.append(users_list[0].id)
```